src/commands/openapi/upload.ts

```diff
@@ -122,7 +130,7 @@ export default class OpenAPIUploadCommand extends BaseCommand<typeof OpenAPIUplo
 
     let filename = specFileTypeIsUrl ? nodePath.basename(specPath) : slugify.default(specPath);
 
-    if (!specFileTypeIsUrl && filename !== specPath) {
+    if (!specFileTypeIsUrl && filename !== specPath && !this.flags['legacy-id']) {
```

Comment:
This warning isn't really relevant if you're using `legacy-id` since you can't use it with `slug`

Reply:
agreed! while you're here, mind also adding this? i don't think it makes sense to emit this warning if `--slug` is passed either

```suggestion
    if (!specFileTypeIsUrl && filename !== specPath && !this.flags['legacy-id'] && !this.flags['slug']) {
```